### DISCUSSION TITLE
Training Load Question?

### DISCUSSION THREAD

**Discussion by Itlansavshos** [philosowaffle/peloton-to-garmin#207] [Q&A] [answered]
Does anyone know if by syncing Peloton data via P2G to Garmin, does Garmin just display that data, or does it factor it into their Training Load analytics use it for training, recovery and all the other stuff it does as if a workout was recorded on a device?

**Comment by philosowaffle** [comment, marked as answer]
Check out near the end of the thread where there are screenshots.: https://github.com/philosowaffle/peloton-to-garmin/issues/114

 Short answer, in most cases the data synced via P2G does count towards Training Status and Load.

**Comment by Itlansavshos** [comment]
Thanks philosowaffle, my mileage seems to vary compared to yours. When I look at Connect (web) -> Reports -> Cycling -> Training Status (or All Activities Training Status) I unfortunately don't get what you do.  I don't get a Training Status at the top, the VO2 chart doesn't update, the Exercise Load seems to update, but the 7 Day Training Load is empty. Also the Training Effect report doesn't populate. 

I noticed your charts were focused on Running, so I was guessing maybe their working cause their tracking data from a device your wearing, but if you switched to Cycling and pure Peloton data I'm guessing your going to get data similar to mine!? Or maybe I need to switch my upload type cause maybe they work for you but not for me because of that? 

But as derekkvedar noted some things do update like TSS and Exercise Load, seems a mixed bag, so I going to focus on what metrics I really need and see what they use and what they don't and go from there!

Thanks so much!!!

**Reply by philosowaffle** [reply]
Attaching some more recent screenshots to help see if we find a pattern it what Garmin is doing.

Things to note:
1. Running is my primary sport but I do not do any Peloton Run workouts.
2. All of my running workouts are recorded on a Fenix 6
3. For some reason, Garmin seems to limit the Training Effect/Status charts to only people with [certain watches](https://support.garmin.com/en-US/?faq=VxKazDQ2mkAmDoQbJriEBA). I don't know why they do this, but I can confirm I never got any of these data points until I upgraded to a Fenix 6 summer of last year.
3. All of my Peloton workouts are synced via P2G, I never record them on my watch

**Connect(web) -> Reports -> Cycling -> Training Status**
<img width="1237" alt="Screen Shot 2022-01-26 at 12 53 25 PM" src="https://user-images.githubusercontent.com/7275041/151229789-e5c4e778-1567-4585-9ca7-08f19be51aa4.png">


Here is a zoomed in view of the Exercise Load chart, I have my cursor hovering over one of my Peloton rides.  Each of those green bars corresponds to a Peloton ride.
<img width="877" alt="Screen Shot 2022-01-26 at 12 53 36 PM" src="https://user-images.githubusercontent.com/7275041/151228360-89f366af-b984-415c-a0eb-d96f550de848.png">


Cycling > TSS/IF Score
<img width="1217" alt="Screen Shot 2022-01-26 at 1 09 31 PM" src="https://user-images.githubusercontent.com/7275041/151230313-39377d36-754c-49e7-bdc8-1e5bd95143ae.png">


In summary the following fields/charts update for me:
✅  Training Status Chart
✅ Exercise Load Chart
✅ 7-day Training Load Chart (and correctly categtorizes by training type aerobic,recovery,anaerobic,etc.)
✅ TSS/IF Score Chart
❌  Training Effect Field on Workout Summary
❌  Cycling VO2 Max

I'm at a loss for why Training Effect does not update given Garmin does correctly categorize the workout type in the 7-Day Load Chart.

I'm also at a loss for the VO2 max according to the [Garmin docs](https://support.garmin.com/en-SG/?faq=MyIZ05OMpu6wSl95UVUjp7). It should be able to calculate it... Although its possible Garmin expects the workout itself to document the VO2 max, in which case P2G would have to figure out an equation to use, I'm not sure that's feasible.

> In order to generate a Cycling VO2 Max estimate, you must record a cycling activity with heart rate and power from a power meter for a minimum of 20 minutes above 70% of your max heart rate. It is recommended that this be performed on an indoor trainer to eliminate variables such as elevation change, drafting, and pacing. Cycling VO2 Max can only be estimated when using the Bike or Indoor Bike profiles. The Mountain Biking activity profile will not generate a VO2 Max.

**Reply by philosowaffle** [reply]
Another interesting article from Garmin on this topic, that I actually hadn't read before:

https://support.garmin.com/en-US/?faq=apva7Th82K6Mm0zY4shzD6

**Comment by Itlansavshos** [comment]
That support article is interesting, the 2nd bullet seems to imply if I'm reading it right that they expect 2 activities per week, but I'm reading (assuming) that as from a Garmin device for this to work (You run with a Fenix 6), so makes sense why its working for you and not myself or danielsfrazier.

The third bullet would seem to indicate they wouldn't use P2G to calc Training Status and such, but from what your showing philosowaffle your seeing it in your report -> cycling -> training status, so its not technically true what their saying ;)

So maybe they need some device based data to sweeten the deal and make this all work, run or cycle, otherwise pure P2G data only gets partial analysis. Especially since they state (You will still need to meet the requirements such as heart rate and power (for cycling activities) along with the above requirements which implies to me at least that you need 2 Garmin based device activities for this to work).

Sounds crazy what I'm typing ;) but I've seen some references in researching that Garmin does a lot of processing and these analytics to be done on the device, their connect platform is mainly for display of data, not post processing, so maybe makes sense that it needs some data from devices, on some interval to make things in the cloud properly work.

**Reply by philosowaffle** [reply]
Yup! Your take aways are similar to mine.

**Comment by philosowaffle** [comment]
Since we're nerding out about this, one more interesting thing, P2G actually attempts to impersonate a Forerunner 945 device (as if the workout was recorded on this device), and that is reflected correctly on Garmin Connect.  I chose this device because it is on the list of Training Status supported devices. But to your observation, I think some of the Training Data is normally calculated on the watch itself, and since I don't know how to calculate/set those fields, they are missing.


![brave_2022-01-26_16-59-05](https://user-images.githubusercontent.com/7275041/151261390-8e144997-a2bb-46e8-8331-cf537e099875.png)

**Reply by danielsfrazier** [reply]
I wonder what would happen if you try to impersonate one of the tacx indoor trainers?

**Reply by philosowaffle** [reply]
You're welcome to try changing it to one of the indoor trainers, but I don't think there's going to be much difference since the current device P2G is using should support all of the Training Status/Load features according to Garmin's product sheets.

If anyone would like to try other devices you can do so today by giving P2G [custom device_info.xml file](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#custom-device-info) (re-reading the P2G docs, it appears this may not be entirely cosmetic as I do think it plays some role in the ability to calculate Training Effect, I'm just not sure how big of a role).  A list of the valid device types can [be found here](https://github.com/philosowaffle/peloton-to-garmin/blob/master/src/Garmin/FitSdk/Dynastream/Fit/Profile/Types/GarminProduct.cs#L25).

**Reply by danielsfrazier** [reply]
I got the idea to change device types because the [link ](https://support.garmin.com/en-US/?faq=apva7Th82K6Mm0zY4shzD6) you posted stated the following: "Only activities uploaded from the Tacx Training app, Zwift, or TrainerRoad will count towards Training Status when you are recording an activity without a Garmin device." 

I changed to TacxTrainingAppWin product id 20533 and I now get VO2 showing up for today's ride. However, the activity details didn't populate with training data but the activity is still listed under the training status like it did for the 945. No new data was posted under "Training Effect" for the cycle activities. Not sure if they require multiple activities. I'm trying to track down someone that has a Tacx or uses Zwift or TrainerRoad to get a fit file that may be of use to you to compare.

Another benefit of changing from 945 to Tacx Training app is these rides may now count towards Garmin Challenges. They have a Tacx Ride to 250 Challenge starting Feb 1. I'll ride again tomorrow to find out.

Side note: Your device list (uploaded to github in June 2021) had the "Venu2plus = 3851;" listed well before it was announced at CES in January 2022. I checked out the latest public SDK device list and the venu2plus was removed from the device list. I'll have to keep an eye on the SDK for hints of future device releases. I'm anxiously awaiting the FR955.

**Reply by philosowaffle** [reply]
> Your device list (uploaded to github in June 2021)

Also not a bad reminder that I should look at pulling in the latest SDK. Maybe there are new options for fields I can set.

**Comment by Itlansavshos** [comment]
That makes sense then why they are calculating a Training Status since you impersonate a 945, otherwise I'm guessing you wouldn't see the rich data you do! 

And to your point "I'm also at a loss for the VO2 max according to the Garmin docs. It should be able to calculate it... Although its possible Garmin expects the workout itself to document the VO2 max, in which case P2G would have to figure out an equation to use, I'm not sure that's feasible." This link confirms it is done on Device. https://support.garmin.com/en-SG/?faq=MyIZ05OMpu6wSl95UVUjp7

This site seems https://sportcoaching.co.nz/how-does-garmin-calculate-vo2-max/ to think they know the formula for calcing VO2max but no idea if its right. ;) Seems like there is a lot of special sauce happening in Garmin on this side of things!
